text/204-golang-bindings.md

```diff
@@ -0,0 +1,655 @@
+---
+feature name: Go Language Bindings
+start date: 2020-07-17
+rfc pr: (leave this empty)
+related issue: <https://github.com/aws/aws-cdk/issues/547>
+---
+
+# Summary
+
+This RFC proposes to support Go language bindings for the CDK.
+
+# Motivation
+
+Go has seen a huge rise in popularity in recent years and is the language of choice for many infrastructure tools, particularly in the container
+world. It has ranked high in recent years on the StackOverflow Developer Survey, especially for such a young language, and with the continued
+popularity of containers and trend towards developers owning their infrastructure, there is a high demand for using Go for infrastructure-as-code
+tooling.
+
+# Design Summary
+
+In order to create the JSII Go language binding, the various types defined in Typescript for the CDK construct library need to be translated into
+equivalent types in Go. The three main types discussed here are Enums, Interfaces, and Classes.
+
+# Detailed Design
```

Comment:
For each section, please add a sub-section that shows the usage from a user perspective.

Reply:
That was the idea behind the code examples (with links to Go playground) -- do you mean you want those extracted into a separate section?

Reply:
The existing examples are great. I just didn't see them consistently in all sections

Reply:
Maybe always "work backwards" from that usage example and then provide the details.